webhooks/run_test.go

```diff
@@ -61,13 +61,15 @@ func run(ctx context.Context, cfg *rest.Config, scheme *runtime.Scheme, opts *en
 		},
 	})
 
-	go func(ctx context.Context, backfillComplete chan struct{}) {
-		podMutator.BackfillPermissions(ctx, backfillComplete)
-	}(ctx, backfillComplete)
+	errChan := make(chan error, 1)
+	go func() {
+		if err := podMutator.BackfillPermissions(ctx); err != nil {
+			errChan <- err
+		}
+	}()
+	go func() {
+		errChan <- err
+	}()
```

Comment:
This will always return the error on L21 (even if it's nil) and ignore the potential error on L67

Reply:
resolved, now returns the error from `mgr.Start(ctx)`, if `mgr.Start(ctx)` fails, it will do so before the potential timeout error on L67, so should be safe for returning the appropriate error now. In the case of no error, `mgr.Start(ctx)` will return nil and the function will be unblocked